ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtil.java

```diff
@@ -553,16 +608,10 @@ public static int bytes3ToInt(Bytes src, int pos) {
     return split_arr;
   }
 
-  /** A helper method for readability. */
-  public static int clamp(int minval, int maxval, int x) {
-    if (x <= minval) return minval;
-    if (x >= maxval) return maxval;
-    return x;
-  }
-
   /** Returns true if the operand is an exact power of two */
-  public static boolean is_power_of_two(long value) {
-    return (value != 0) && (Long.lowestOneBit(value) == Long.highestOneBit(value));
+  public static boolean is_power_of_two(UnsignedLong value) {
+    return (value.compareTo(UnsignedLong.ZERO) != 0)
+        && (Long.lowestOneBit(value.longValue()) == Long.highestOneBit(value.longValue()));
```

Comment:
We should write up an issue to confirm that `Long.lowestOneBit` and `Long.highestOneBit` work as expected for large/2s complement negative numbers.

Reply:
yessir.... i plan on testing this before submitting this one for review

Reply:
I don't think it does work:
```
Long.lowestOneBit(-16L)
16
Long.highestOneBit(-16L)
-9223372036854775808
```